source/apiVolontaria/volunteer/serializers.py

```diff
@@ -98,65 +107,114 @@ def create(self, validated_data):
-        address_data = validated_data['address']
-        country_data = validated_data['address']['country']
-        state_province_data = validated_data['address']['state_province']
 
-        address_data['country'] = country_data['iso_code']
-        address_data['state_province'] = state_province_data['iso_code']
-
-        try:
-            cell_address = Address.objects.get(**address_data)
-        except Address.DoesNotExist:
-            try:
-                cell_address = Address()
-                cell_address.__dict__.update(validated_data['address'])
-                cell_address.country, created = Country.objects.get_or_create(
-                    name__iexact=country_data['name'],
-                    iso_code__iexact=country_data['iso_code'],
-                    defaults={
-                        'iso_code': country_data['iso_code'],
-                        'name': country_data['name'],
-                    },
-                )
-            except IntegrityError as err:
-                if 'UNIQUE constraint failed' in err.args[0]:
+        list_manager = list()
+        if managers:
+            for manager in managers:
+                try:
+                    user = User.objects.get(id=manager)
+                    list_manager.append(user)
+                except Exception as err:
                     error = {
                         'message': (
-                            "A Country with that iso_code already exists"
+                            "Unknown user with this ID"
                             if err.args else "Unknown Error"
                         )
                     }
                     raise serializers.ValidationError(error)
-            try:
-                cell_address.state_province, created = (
-                    StateProvince.objects.get_or_create(
-                        iso_code__iexact=state_province_data['iso_code'],
-                        name__iexact=state_province_data['name'],
-                        country=cell_address.country,
-                        defaults={
-                            'iso_code': state_province_data['iso_code'],
-                            'name': state_province_data['name'],
-                        },
-                    )
-                )
-                cell_address.save()
-            except IntegrityError as err:
-                if 'UNIQUE constraint failed' in err.args[0]:
+
+        cell.save()
+        for user in list_manager:
+            cell.managers.add(user)
+        return cell
+
+    def update(self, instance, validated_data):
+        if 'name' in validated_data.keys():
+            instance.name = validated_data['name']
+        if 'managers' in validated_data.keys():
+            # Remove all managers of the cell
+            instance.managers.clear()
+
+            for manager in validated_data['managers']:
+                try:
+                    user = User.objects.get(id=manager)
```

Comment:
the output of manager is for example : 
`{'id': 8, 'username': 'root', 'first_name': '', 'last_name': '', 'email': 'a@b.com'}`

so this line should be : 
`user = User.objects.get(id=manager['id'])`

Reply:
I agree with you the output of a manager is :
```
{'id': 8, 'username': 'root', 'first_name': '', 'last_name': '', 'email': 'a@b.com'}
```

But his input is just one or multiple IDs:
```
managers: [
    31,
]
```

Reply:
Okay, make sense.